Initial commit for kinesis_encrypt_stream

lytran2000 opened the pull request:
may you review and approve

kshrutik commented on remediation_worker/jobs/kinesis_encrypt_stream/kinesis_encrypt_stream.py:
```diff
@@ -0,0 +1,173 @@
+                    stream_name
+                )
+            sys.exit(1)
+        except:
```
Can you please log the exact error returned by AWS instead of giving the generic error. It will be more helpful and informative.

lytran2000 replied:
Changed the exception to ClientError and log all AWS error accordingly.

kshrutik commented on remediation_worker/jobs/kinesis_encrypt_stream/kinesis_encrypt_stream.py:
```diff
@@ -0,0 +1,173 @@
+        # If kinesis data stream is not encrypted
+        if encryption_type == "NONE":
+            self.enable_encryption(client, stream_name)
+        # If encryption is enabled
```
There is no exit or return statement after enabling the encryption. You can write only one return for both if and elif block. i.e write the return statement after if-elif-else block. Something like this -

```
def remediate(self, stream_name, client, region):
        """Encrypts unencrypted kinesis data streams.
        Args:
            stream_name ([str]): The name of the kinesis stream.
            client: Instance of the AWS boto3 client.
        """

        logging.info(
            "Beginning kinesis data stream encryption evaluation."
        )

        # Check to see if kinesis data stream is encrypted
        encryption_type = self.check_encryption(client, stream_name)

        # If kinesis data stream is not encrypted
       if encryption_type == "NONE":
            self.enable_encryption(client, stream_name)
        # If encryption is enabled
        # We have nothing to do.
        # log and exit
        elif encryption_type == "KMS":
            logging.info(
                "Kinesis data stream %s is already encrypted, taking no action.",
                stream_name
            )
        # If anything else returns,
        # log and exit
        else:
            logging.error(
                "An unknown encryption type returned for kinesis data stream %s. Exiting.",
                stream_name
            )
            sys.exit(1)
        return 0
```

kshrutik commented on remediation_worker/jobs/kinesis_encrypt_stream/kinesis_encrypt_stream.py:
```diff
@@ -0,0 +1,173 @@
+                format(found_error.response['Error']['Code'])
+            )
+            sys.exit(1)
+
```
Can you add another except block and log the error returned by AWS other than ResourceNotFoundException.

goshkis replied:
can't  help but notice that format() is meaningless here

kshrutik commented on test/unit/test_kinesis_encrypt_stream.py:
```diff
@@ -0,0 +1,39 @@
+        assert "stream_name" in param
+        assert param["stream_name"] == "kinesis-stream"
+        assert "region" in param
+        assert param["region"] == "us-east-1"
```
Can you please add a test case for remediate method as well?

lytran2000 commented:
I have 
1. updated the README file
2. updated kinesis_encrypt_stream.py for proper return code and exception handling
3. added more test cases for remediation
May you help to review?

lytran2000 commented:
Just noticed I had to resolve another README file.  Did another push.

lytran2000 commented:
Removed the unnecessary format in kinesis_encrypt_stream.py

kshrutik commented on remediation_worker/jobs/kinesis_encrypt_stream/kinesis_encrypt_stream.py:
```diff
@@ -0,0 +1,187 @@
+            return 1
+        except Exception as e:
+            error = "Receiving exception {0} for kinesis data stream {1}".format(str(e), stream_name)
+            return 1
```
log the error - 
logging.error(error)

lytran2000 commented:
Hi Shrutika,
I have resolved the README and added the missing error logging.  May you review?
